source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,538 @@
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Cross Contract Communication
+
+Level of the tutorial: `Advanced`
+
+Time to complete the tutorial: `45 - 60 Minutes`
+
+Important topics to read before going further:
+
+- [getting started](../../../developers/writing-onchain-code/getting-started.md)
+- [writing on-chain code](../../../developers/writing-onchain-code/simple-contract.md)
+- [calling contracts](../../../developers/writing-onchain-code/calling-contracts.md)
+
+This will assure that you have needed tools and base knowledge to complete this tutorial. If not everything was clear while reading these sections, don't worry as we will be doing a full walkthrough step by step.
+
+## Outline of the tutorial {#outline}
+
+This tutorial covers all possible variations of cross contract communication. While it not necessary when writing a simple contract which doesn't have to interact with other deployed contracts it is nontheless important to understand the ways to achieve this feat.
+
+On a very high level a smart contract behaves like a class in object oriented programming. This would suggest, that the principles for object oriented programming also apply for smart contract development. What interests us the most is the responsibility separation of classes. On a conceptual level for very simple functionalities there is no need to separate the logic between more smart contracts. It is advised to think about it and preplan the development. A smart contract on Casper Network can be always updated, but for more complex scenarios there is a need to fully understand how the project can be segrageted into smaller units of logical work.
```

Comment:
"this would suggest" is vague. You're not committing to a statement there. Try something like "THis means that the principles of good OO development may also be applied in contract development"

Reply:
Agreed. Deleted this section as it was too vague and was confusing to read.